.github/workflows/pkgci_test_amd_mi300.yml

```
@@ -19,7 +19,7 @@ on:
 
 jobs:
   test_mi300:
```

This job could be renamed to mi325

```suggestion
  test_mi325:
```

---

Oh, and we can rename this entire file too, along with the references in https://github.com/iree-org/iree/blob/main/.github/workflows/pkgci.yml

---

Cool, @deedongala can you re-name these

---

I've renames all references.